### PR TITLE
Unnecessary focus on the Add payment method button

### PR DESCRIPTION
**Pull request by allroundexperts**
### Details

The css selector :focus-visible is responsible for showing up the outline. When the selector :focus-visible is applied to an element is largely dependent browser implementation. As always, the problematic browser is Safari. Safari restores focus on an element even if an event is a click. On Chrome and other browser, focus is restored if event type is keypress ie closing modal by keyboard.

### Fixed Issues

$ https://github.com/Expensify/App/issues/9248

### Tests

1. Go to Settings
2. Go to Payments
3. Click on the Add payment method button
4. Click anywhere on the Payments page

Make sure that there is no focus on the Add payment method button ie
image

- [x] Verify that no errors appear in the JS console

#### Contributor (PR Author) Checklist
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [ ] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose and it is
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn’t already exist
    - [ ] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.

<details>
<summary><h4>PR Reviewer Checklist</h4></summary>

- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn’t already exist
    - [ ] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.

</details>

### QA Steps

1. Go to Settings
2. Go to Payments
3. Click on the Add payment method button
4. Click anywhere on the Payments page

Make sure that there is no focus on the Add payment method button ie
image

- [x] Verify that no errors appear in the JS console

#### Web
https://user-images.githubusercontent.com/30054992/173203253-70cb00d7-0f62-44a0-bb90-3bc17955f1a2.mov

#### Mobile Web

https://user-images.githubusercontent.com/30054992/171935670-42737e6f-02f4-4fbd-860a-16ad9e8413e9.mp4

#### Android
https://user-images.githubusercontent.com/30054992/173203299-6e8b4f68-12d0-4d5f-9b6b-e37489f14087.mov

#### iOS
https://user-images.githubusercontent.com/30054992/173203649-20704138-1ce6-45d4-8173-71d7077be0fa.mp4

#### Desktop
https://user-images.githubusercontent.com/30054992/173203718-0e4b4738-b8be-427c-a38e-6004a75f4ae6.mov

### CODE REVIEW

**Comment by rushatgabhane** [comment]
@allroundexperts thanks for raising the PR so quickly!

Could you please add tests for this comment https://github.com/Expensify/App/issues/9248#issuecomment-1147806169, and add screenshots for all platforms.

**Comment by allroundexperts** [comment]
> @allroundexperts thanks for raising the PR so quickly!
> 
> Could you please add tests for this comment [#9248 (comment)](https://github.com/Expensify/App/issues/9248#issuecomment-1147806169), and add screenshots for all platforms.

Are you talking about unit tests @rushatgabhane?

**Comment by rushatgabhane** [comment]
@allroundexperts sorry that I wasn't clear. I was referring to test steps in the PR description.

**Comment by rushatgabhane** [comment]
@allroundexperts just waiting on this https://github.com/Expensify/App/pull/9348#issuecomment-1149227462

**Comment by allroundexperts** [comment]
@rushatgabhane Updated.

**Review by rushatgabhane — DISMISSED** [review]
@thienlnam LGTM!
Also, the user can navigate through menus and buttons by clicking and then using keyboard shortcuts and vice versa. I verified it for a few other buttons too.

https://user-images.githubusercontent.com/29673073/173363129-bad14546-7a6c-4c76-ab0f-33fda05e370d.mov

**Comment by rushatgabhane** [comment]
<details>
<summary>PR Reviewer Checklist</summary>

- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [ ] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn’t already exist
    - [ ] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.

</details>

**Comment by rushatgabhane** [comment]
Found a console error which is unrelated to the PR. https://expensify.slack.com/archives/C01GTK53T8Q/p1655128339662359

**Comment by rushatgabhane** [comment]
> make sure to check that the user can navigate through menus and buttons by clicking and then using keyboard shortcuts

@allroundexperts can you please add some steps for this in the Tests and QA section of this PR.

File: src/components/Button.js [hunk 1]
```diff
@@ -233,6 +233,9 @@ class Button extends Component {
         return (
             <Pressable
                 onPress={(e) => {
+                    if (e.type === 'click') {
+                        e.currentTarget.blur();
+                    }
```
Comment (thienlnam) [inline src/components/Button.js:238]: Style - add another line break before the next if statement

File: src/components/MenuItem.js [hunk 2]
```diff
@@ -48,6 +48,9 @@ const MenuItem = props => (
             if (props.disabled) {
                 return;
             }
+            if (e.type === 'click') {
```
Comment (thienlnam) [inline src/components/MenuItem.js:51]: same as above

**Review by thienlnam — CHANGES_REQUESTED** [review]

File: src/components/MenuItem.js (hunk 2 above)
Comment (allroundexperts) [reply src/components/MenuItem.js:52]: Done.

File: src/components/Button.js (hunk 1 above)
Comment (allroundexperts) [reply src/components/Button.js:238]: Done.

**Review by thienlnam — APPROVED** [review]

File: src/components/Button.js [hunk 3]
```diff
@@ -233,6 +233,10 @@ class Button extends Component {
         return (
             <Pressable
                 onPress={(e) => {
+                    if (e.type === 'click') {
```
Comment (thienlnam) [inline src/components/Button.js:236]: Sorry, let's also check if e.type exists so

if (e.type && e.type === 'click')

Same with below

**Review by thienlnam — CHANGES_REQUESTED** [review]

File: src/components/Button.js (hunk 3 above)
Comment (allroundexperts) [reply src/components/Button.js]: Handled.

Comment (thienlnam) [reply src/components/Button.js]: The way I mentioned is preferred

Comment (allroundexperts) [reply src/components/Button.js]: Why though? If `e` is undefined, what you've suggested will throw an error.

Comment (thienlnam) [reply src/components/Button.js]: fair enough, let's just check if e exists then

Comment (allroundexperts) [reply src/components/Button.js]: How about `if (e?.type === 'click')` ?

Comment (allroundexperts) [reply src/components/Button.js]: `if (e.type && e.type === 'click')` is exactly same as `e.type === 'click`. Not sure what you're trying to do here.

Comment (thienlnam) [reply src/components/Button.js]: if e && e.type === 'click'

Comment (allroundexperts) [reply src/components/Button.js]: Fixed.

**Review by thienlnam — APPROVED** [review]
Awesome, thanks for accommodating those changes

**Pull request merged**